File: solutions/Java/reverse_bits_iterate.java
public class ReverseBitsIterate {
    public static int reverseBitsIterate(int n) {
        int result = 0;
        for (int i = 0; i < 32; i++) {
            result = (result << 1) | (n & 1);
            n >>>= 1;
        }
        return result;
    }

    public static void main(String[] args) {
        System.out.println(reverseBitsIterate(43261596));
    }
}

File: solutions/Rust/reverse_bits_iterate.rs
fn reverse_bits_iterate(mut n: u32) -> u32 {
    let mut result = 0;
    for _ in 0..32 {
        result = (result << 1) | (n & 1);
        n >>= 1;
    }
    result
}

fn main() {
    println!("{}", reverse_bits_iterate(43261596));  // 964176192
}
